src/test/java/com/firebolt/jdbc/resultset/FireboltResultSetMetaDataTest.java

```diff
@@ -93,19 +99,57 @@ void shouldReturnColumnClassName() throws SQLException {
 
 	@Test
 	void shouldReturnTrueWhenColumnIsCaseSensitiveAndFalseOtherwise() throws SQLException {
-		FireboltResultSetMetaData fireboltResultSetMetaData = getMetaData();
+		ResultSetMetaData fireboltResultSetMetaData = getMetaData();
 		assertTrue(fireboltResultSetMetaData.isCaseSensitive(1));
 		assertFalse(fireboltResultSetMetaData.isCaseSensitive(2));
 	}
 
 	@Test
-	void isReadOnly() throws SQLException {
-		assertTrue(getMetaData().isReadOnly(1));
+	void trivialMarkers() throws SQLException {
+		ResultSetMetaData md = getMetaData();
+		assertTrue(md.isReadOnly(1));
+		assertFalse(md.isAutoIncrement(1));
+		assertTrue(md.isSearchable(1));
+		assertFalse(md.isCurrency(1));
+		assertEquals(80, md.getColumnDisplaySize(1));
+		assertEquals("", md.getSchemaName(1));
+		assertFalse(md.isWritable(1));
+		assertFalse(md.isDefinitelyWritable(1));
+	}
+
+	@ParameterizedTest
+	@ValueSource(ints = {-1, 0, 4})
+	void wrongIndex(int column) {
+		ResultSetMetaData md = getMetaData();
+		assertFunction(md::isNullable, column);
+		assertFunction(md::isSigned, column);
+		assertFunction(md::getColumnLabel, column);
+		assertFunction(md::getColumnName, column);
+		assertFunction(md::getPrecision, column);
+		assertFunction(md::getScale, column);
+		assertFunction(md::getTableName, column);
+		assertFunction(md::getCatalogName, column);
+		assertFunction(md::getColumnType, column);
+		assertFunction(md::getColumnTypeName, column);
+		assertFunction(md::getColumnClassName, column);
+		assertFunction(md::isCaseSensitive, column);
+		assertFunction(md::isAutoIncrement, column);
+		assertFunction(md::isSearchable, column);
+		assertFunction(md::isCurrency, column);
+		assertFunction(md::getColumnDisplaySize, column);
+		assertFunction(md::getSchemaName, column);
+		assertFunction(md::isReadOnly, column);
+		assertFunction(md::isWritable, column);
+		assertFunction(md::isDefinitelyWritable, column);
+	}
+
+	private <R> void assertFunction(CheckedFunction<Integer, R> function, int column) {
```

Comment:
nit: `assertFunctionThrows` , otherwise it's reading like it's verifying success unless you read the implementation.

Reply:
Good point. Renamed.